libavfilter/dnn/dnn_backend_native_layer_mathbinary.c

```diff
@@ -102,95 +160,23 @@ int dnn_execute_layer_math_binary(DnnOperand *operands, const int32_t *input_ope
     dst = output->data;
 
     switch (params->bin_op) {
-    case DMBO_SUB:
-        if (params->input0_broadcast) {
-            for (int i = 0; i < dims_count; ++i) {
-                dst[i] = params->v - src[i];
-            }
-        } else if (params->input1_broadcast) {
-            for (int i = 0; i < dims_count; ++i) {
-                dst[i] = src[i] - params->v;
-            }
-        } else {
-            const DnnOperand *input1 = &operands[input_operand_indexes[1]];
-            const float *src1 = input1->data;
-            for (int i = 0; i < dims_count; ++i) {
-                dst[i] = src[i] - src1[i];
-            }
-        }
+    case DMBO_SUB:  
+        mathbin2(SUB_fun, params, src, dst, dims_count, operands, input_operand_indexes); 
         return 0;
-    case DMBO_ADD:
-        if (params->input0_broadcast || params->input1_broadcast) {
-            for (int i = 0; i < dims_count; ++i) {
-                dst[i] = params->v + src[i];
-            }
-        } else {
-            const DnnOperand *input1 = &operands[input_operand_indexes[1]];
-            const float *src1 = input1->data;
-            for (int i = 0; i < dims_count; ++i) {
-                dst[i] = src[i] + src1[i];
-            }
-        }
+    case DMBO_ADD: 
+        mathbin1(ADD_fun, params, src, dst, dims_count, operands, input_operand_indexes);  
```

Comment:
why we use different function for sub and add?

Reply:
Different from add, sub, realdiv and floormod needs to consider the order of two input data.

Reply:
but, for current code,  add and sub uses different function, right?
>Different from add, sub, realdiv and floormod needs to consider the order of two input data.

Not familiar with this. But tf's realdiv has no order. https://www.w3cschool.cn/tensorflow_python/tensorflow_python-fi9u2hcl.html

Reply:
> but, for current code, add and sub uses different function, right?
> 
> > Different from add, sub, realdiv and floormod needs to consider the order of two input data.
> 
> Not familiar with this. But tf's realdiv has no order. https://www.w3cschool.cn/tensorflow_python/tensorflow_python-fi9u2hcl.html

yes, sub needs to consider the order, and add does not.

Reply:
> > but, for current code, add and sub uses different function, right?
> > > Different from add, sub, realdiv and floormod needs to consider the order of two input data.
> > 
> > 
> > Not familiar with this. But tf's realdiv has no order. https://www.w3cschool.cn/tensorflow_python/tensorflow_python-fi9u2hcl.html
> 
> This function takes into account the existence of constants in the input. Because when there is a constant, we do not need to traverse every position of the input data for the input constant, so in this case, he will use input0_ broadcast， input1_broadcast represents the input position of the constant, and params->v is the value of the constant. In this case, there are two different orders for sub functions.
If there is no constant, input0_broadcast, input1_broadcast, If the are both 0, then for sub and add, there is no need to consider the order problem, only need to execute the function according to the input order.

Reply:
> This function takes into account the existence of constants in the input. Because when there is a constant, we do not need to traverse every position of the input data for the input constant, so in this case, he will use input0_ broadcast， input1_broadcast represents the input position of the constant, and params->v is the value of the constant. In this case, there are two different orders for sub functions.
> If there is no constant, input0_broadcast, input1_broadcast are both 0, then for sub and add, there is no need to consider the order problem, only need to execute the function according to the input order.